.github/workflows/ci-commit.yml

```diff
@@ -0,0 +1,345 @@
+name: ci
+
+on:
+  workflow_dispatch:
+  push:
+    branches:
+      - master
```

Comment:
I do sort of question this, as checking once things have hit master is a bit late...

Reply:
We can't have self-hosted runners in a PR environment.  Also, all of this should move to a separate repo with added protections on who can edit and call the CI steps on the workflow.